packages/protocol/contracts/stability/Exchange.sol

```diff
@@ -168,11 +161,11 @@ contract Exchange is IExchange, Initializable, Ownable, UsingRegistry {
     uint256 buyTokenBucket;
     (buyTokenBucket, sellTokenBucket) = getBuyAndSellBuckets(sellGold);
 
-    uint256 x = spread.denominator.sub(spread.numerator).mul(sellAmount);
-    uint256 numerator = x.mul(buyTokenBucket);
-    uint256 denominator = sellTokenBucket.mul(spread.denominator).add(x);
+    int256 reducedSellAmount = getReducedSellAmount(sellAmount);
+    int256 numerator = reducedSellAmount.multiply(toFixed(buyTokenBucket));
+    int256 denominator = toFixed(sellTokenBucket).add(reducedSellAmount);
```

Comment:
does this means that buckets can't be larger than `uint128`? And doesn't this means that if we ever have a bucket that large, we will only realize that too late?